### PR TITLE
Bnscli fee

### PR DESCRIPTION
**Pull request by husio**
A new `cmd/bnscli with-fee` command implemented as requested in #742

### CODE REVIEW

File: cmd/bnscli/clitests/cli_test.go [hunk 1]
```diff
@@ -112,6 +112,25 @@ func fakeTendermintServer(t testing.TB) *httptest.Server {
 					}
 				}
 			`)
+		case r.Method == "GET" && r.URL.Path == "/abci_query":
+			// Handle all data queries. Values are JSON encoded.
+
+			switch r.URL.Query().Get("data") {
+			case `"_c:cash"`:
```
Comment (ethanfrey) [inline cmd/bnscli/clitests/cli_test.go]: nice mock sever

File: cmd/bnscli/clitests/send_tokens_with_default_fee.test.gold [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+	"fees": {
+		"fees": {
+			"whole": 17,
```
Comment (ethanfrey) [inline cmd/bnscli/clitests/send_tokens_with_default_fee.test.gold]: Nice to have both tests... explicit fee and server-defined product fee

File: cmd/bnscli/cmd_cash.go [hunk 3]
```diff
@@ -42,3 +45,92 @@ destination account.
 	_, err := writeTx(output, tx)
 	return err
 }
+
+func cmdWithFee(input io.Reader, output io.Writer, args []string) error {
```
Comment (ethanfrey) [inline cmd/bnscli/cmd_cash.go]: Yes, logic here looks very straight-forward and correct

File: cmd/bnscli/cmd_msgfee.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/x/msgfee"
+)
+
+func msgfeeConf(nodeUrl string, msgPath string) (*coin.Coin, error) {
+	queryUrl := nodeUrl + "/abci_query?path=%22/%22&data=%22msgfee:" + url.QueryEscape(msgPath) + "%22"
```
Comment (ethanfrey) [inline cmd/bnscli/cmd_msgfee.go]: You really don't want to use tendermint client, right?
It seems a bit much to create these queries entirely by hand, but it works and if what you like, fine.

Comment (husio) [reply cmd/bnscli/cmd_msgfee.go]: In my opinion, tendermint client is not good enough. Using plain HTTP has for me the advantage that anyone read and understands it instantly. It is a basic HTTP request. Mocking is straightforward as well.

I would consider using a client when request/response goes beyond simple HTTP and even then I would honestly consider writing a custom client :see_no_evil:

**Pull request merged**